src/Zynga/Framework/Cache/V2/Driver/InMemory.hh

```diff
@@ -48,20 +48,37 @@ class InMemory extends DriverBase implements MemcacheDriverInterface{
     $valueSet = self::$data->get($key);
     return $valueSet === null ? false : true;
   }
-  
+
   public function directDelete(string $key): bool {
     $data = self::$data->get($key);
 
-    if($data === null){
+    if ($data === null) {
       return false;
     }
 
     self::$data->remove($key);
 
     return true;
   }
-  
-  public function connect(): bool  {
+
+  /**
+   * Equivalent to directAdd for the InMemory driver because
+   * Map's set and add methods are equivalent.
+   */
+  public function directSet(
+    string $key,
+    mixed $value,
+    int $flags = 0,
+    int $ttl = 0,
+  ): bool {
+    return $this->directAdd($key, $value, $flags, $ttl);
```

Comment:
Add and set should be different. Looking at your code, i just realized i messed up directAdd. Add should only work when there is no value. I can fix add on master, can you change set to not call add and directly add data to self::$data. 

You can make it similar to set. 
```
self::$data->set($key, $value);
$value = self::$data->get($key);	    
return $value === null ? false : true;
```	     

Also need to add some tests in InMemoryTest.hh